autogpt_platform/backend/backend/server/v2/store/db.py

```diff
@@ -588,6 +604,134 @@ async def create_store_submission(
         ) from e
 
 
+async def create_store_version(
+    user_id: str,
+    agent_id: str,
+    agent_version: int,
+    store_listing_id: str,
+    name: str,
+    video_url: str | None = None,
+    image_urls: list[str] = [],
+    description: str = "",
+    sub_heading: str = "",
+    categories: list[str] = [],
+    changes_summary: str = "Update Submission",
+) -> backend.server.v2.store.model.StoreSubmission:
+    """
+    Create a new version for an existing store listing
+
+    Args:
+        user_id: ID of the authenticated user submitting the version
+        agent_id: ID of the agent being submitted
+        agent_version: Version of the agent being submitted
+        store_listing_id: ID of the existing store listing
+        name: Name of the agent
+        video_url: Optional URL to video demo
+        image_urls: List of image URLs for the listing
+        description: Description of the agent
+        categories: List of categories for the agent
+        changes_summary: Summary of changes from the previous version
+
+    Returns:
+        StoreSubmission: The created store submission
+    """
+    logger.debug(
+        f"Creating new version for store listing {store_listing_id} for user {user_id}, agent {agent_id} v{agent_version}"
+    )
+
+    try:
+        # First verify the listing belongs to this user
+        listing = await prisma.models.StoreListing.prisma().find_first(
+            where=prisma.types.StoreListingWhereInput(
+                id=store_listing_id, owningUserId=user_id
+            )
+        )
+
+        if not listing:
+            logger.warning(
+                f"Store listing not found for user {user_id}: {store_listing_id}"
+            )
+            raise backend.server.v2.store.exceptions.ListingNotFoundError(
+                f"Store listing not found. User ID: {user_id}, Listing ID: {store_listing_id}"
+            )
+
+        # Verify the agent belongs to this user
+        agent = await prisma.models.AgentGraph.prisma().find_first(
+            where=prisma.types.AgentGraphWhereInput(
+                id=agent_id, version=agent_version, userId=user_id
+            )
+        )
```

Comment:
Isn't this check excessive? information in `agent` is not used at all, and we've been querying the listing based on the owningUserId, why add another db query for this?

Reply:
wouldn't it otherwise be possible for a user to submit an agent they don't own? 

`owningUserId` tells us who owns the `StoreListing`, but unless we enforce this here, they could list an agent they do not own as far as I can tell

Reply:
ahh I see, got it :+1: